Added (PRIVATE) to summary line of private docstrings in Bio.Restriction.Restriction.py

Amrithasuresh opened the pull request:
@peterjc Please have a look at the changes. Let me know if there are any mistakes. #1466

Thank you.

I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

I have read the ``CONTRIBUTING.rst`` file and understand that AppVeyor and
TravisCI will be used to confirm the Biopython unit tests and ``flake8`` style
checks pass with these changes.

I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

peterjc commented:
Changes look fine - are you sure you don't want to be thanked by name as a contributor?

peterjc commented:
I agree with the aspiration - even private docstrings benefit from short one line summaries. It is more work to adjust those, but on the bright side there are easy to detect later with a PEP8 tool.

Amrithasuresh commented:
>>are you sure you don't want to be thanked by name as a contributor?

@peterjc yes I am sure. I don't want.

It seems some checks were not successful. Do I need to fix anything? Let me know. Thank you.

peterjc commented on Bio/Restriction/Restriction.py:
```diff
@@ -2323,14 +2323,14 @@ def _boundaries(self, start, end):
 
     def _test_reverse(self, start, end, site):
-        """Test if site is between end and start (for circular sequences).
+        """Test if site is between end and start (for circular sequences) (PRIVATE).
```
To try to keep this line under 80 characters, how about:

``Test if site is between end and start (for circular sequences) (PRIVATE).``

-->

``Test if site between end & start (for circular sequences) (PRIVATE).``

Amrithasuresh commented:
@peterjc I have few queries

1. Do I need to consider only the edited lines (Added PRIVATE) to check if it is under 80 character. Is that right?

2. For example in the file Bio/Restriction/Restriction.py line 2333 below

                """ Test if site is between end and start (for circular sequences).
 
                 Internal use only.
                 """

If the character length exceeds 80 characters in a line then I can move to the next line. Like below. Is that right?

                """ Test if site is between end and start (for circular sequences) 
               (PRIVATE).
 
                 Internal use only.
                 """

3. I already did a commit for the pull request. So I  have to do another commit with a commit message. Is that right?

peterjc commented:
1. Yes.
2. No. The docstring structure requires a one line summary, then a blank line, then more text. See https://www.python.org/dev/peps/pep-0257/#one-line-docstrings Here if you really can't see how to shorten the one line summary, then having a long line is the only solution.
3. Another commit is fine.

Amrithasuresh commented:
@peterjc there was only one line that was more than 80 character line (that you mentioned it). I fixed it as per your suggestion and committed.

I had some issues

    To https://github.com/Amrithasuresh/biopython.git
    ! [remote rejected]     B1 -> B1 (pre-receive hook declined)
    error: failed to push some refs to 'https://github.com/Amrithasuresh/biopython.git'

However, when I updated master branch and then again I pushed. It seems it is successfully fixed. Let me know if this is ok.

Thank you.

peterjc commented:
I'm not sure about that git issue, a quick google suggests it can be an authentication problem - but it seems to have worked now.

The change looks sensible, I'm just waiting for the automated tests to finish.

peterjc commented:
I'm sorry, it looks like I missed this in the #1466 discussion, and these files got fixed as part of #1749 which I just merged (even though you did this work first). Apologies for overlooking it.
